src/pages/RecipeForm/hooks/useUpdateRecipeMutation.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  UseMutationResult,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { updateRecipe } from "../api";
+import type { RecipeUpdate } from "../../../types";
+import snakify from "snakify-ts";
+
+const useUpdateRecipeMutation = (): UseMutationResult<
+  void,
+  Error,
+  RecipeUpdate
+> => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async (recipe: RecipeUpdate) => {
+      const { id, ...recipeDetails } = recipe;
+      await updateRecipe(recipe.id, snakify(recipeDetails));
+    },
+    onSuccess: () => queryClient.invalidateQueries({ queryKey: ["recipes"] }),
+  });
```

Comment:
Wrapping my head around. Noted to learn about mutation and query thing too 🙄 
Will message you in case I find any trouble. 🤯

Reply:
No worries, we used GraphQL as our BFF at Zoopla so I'm more used to using a client for that, rather than this library!